src/test/java/test/DynamicGraphTest.java

```diff
@@ -158,6 +166,10 @@ public void test3() {
           assertFreeNodesEquals(dg);
 
           dg.setStatus(b, Status.FINISHED);
+          // TODO https://github.com/cbeust/testng/issues/1380
+          // a and c should not be free at the same time:
+          // if c -> b -> a then an explicit edge should exist from c to a.
```

Comment:
@juherr - I wanted to ask this on #1380 itself.. 

When you say 

> a and c should not be free at the same time: if c -> b -> a 

What does this mean ? Does it mean method `c` directly depends on method `b` and in-directly depends on method `a` whereas method `b` has ONLY a direct dependency on `a`.